sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -246,11 +246,50 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     arr
   }
 
+  private def childrenTheSame(
+      originalChildren: IndexedSeq[BaseType], newChildren: IndexedSeq[BaseType]): Boolean = {
+    val size = originalChildren.size
+    var i = 0
+    var childrenTheSame = true
+    while (i < size && childrenTheSame) {
```

Comment:
can we use early return in the loop body?

(it seems more readable to me)

Reply:
early return in loop is not recommended in Scala...

Reply:
And also `break` in Scala doesn't translate to Java break, throws an exception and is expensive.

Reply:
Maybe I'm missing sth., but early return is preferred in the style guide?
https://github.com/databricks/scala-style-guide#return-statements